zerver/lib/actions.py

```diff
@@ -3078,6 +3078,26 @@ def check_update_message(
     if not user_profile.realm.allow_message_editing:
         raise JsonableError(_("Your organization has turned off message editing"))
 
+    if (
+        message.recipient.type == Recipient.STREAM
+        and not Stream.objects.filter(id=message.recipient.type_id, deactivated=False).exists()
+    ):
+        raise JsonableError(_("Stream does not exist"))
+
+    if message.recipient.type == Recipient.PERSONAL:
+        recipient_user = UserProfile.objects.get(id=message.recipient.type_id)
+        if not recipient_user.is_active:
+            raise JsonableError(
+                _("'{email}' is no longer using Zulip.").format(email=recipient_user.email)
+            )
+
+    if message.recipient.type == Recipient.HUDDLE:
+        user_ids = list(get_huddle_user_ids(message.recipient))
+        # Remove the acting user from the list because the user must be active to edit the message.
+        user_ids.remove(user_profile.id)
+        if not UserProfile.objects.filter(id__in=user_ids, is_active=True).exists():
+            raise JsonableError(_("Users in this huddle are no longer using Zulip"))
```

Comment:
Not sure whether we want to show email of users who are deactivated here like we do for a PM with one user.

Reply:
I don't think we need to do that.